doc/contributing/coding-conventions.chapter.md

```diff
@@ -196,7 +196,23 @@
   As an exception, an explicit conditional expression with null can be used when fixing a important bug without triggering a mass rebuild.
   If this is done a follow up pull request _should_ be created to change the code to `lib.optional(s)`.
 
-- Arguments should be listed in the order they are used, with the exception of `lib`, which always goes first.
+- The order of elements in the argument set must follow the convention below:
+
+  01. `lib`;
+  02. `stdenv`;
+  03. builders (`buildPythonPackage`, `mkDerivation`, `buildGoModule`);
```

Comment:
"builders" needs to be defined here, does `rustPlatform` count? what about `makeRustPlatform`? what about `python3` if you are just using `python3.pkgs.buildPythonApplication`?

Reply:
> "builders" needs to be defined here, does `rustPlatform` count? what about `makeRustPlatform`?

Now I am a bit confused. Is `rustPlatform` something similar to `stdenv`?
There are other `languagePlatforms` similar to it, so I can compare?

For now I am inclined to include them here too.

> what about `python3` if you are just using `python3.pkgs.buildPythonApplication`?

This is still `python3`. So, item 6.

I believe this is consistent because you are not seeking for a typical builder, but by the Python-ecosystem-provided one.

Reply:
`rustPlatform` is essentially a wrapper for `cargo`, `rustc`, `cargo-auditable`, and `stdenv`, and exposes things like `buildRustPackage` (a builder, which the most commonly directly used), `fetchCargoTarball` (fod fetcher that takes a `src`), and `rustLibSrc` (derivation)

Reply:
Basically a stdenv with rusteroids.
Fine. Let's lump them.